python/lsst/ts/standardscripts/auxtel/offset_ataos.py

```diff
@@ -0,0 +1,155 @@
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+__all__ = ["OffsetATAOS"]
+
+import yaml
+from lsst.ts.observatory.control.auxtel import ATCS, ATCSUsages
+
+from lsst.ts import salobj
+
+
+class OffsetATAOS(salobj.BaseScript):
+    """Perform an ATAOS offset.
+
+    Parameters
+    ----------
+    index : `int`
+        Index of Script SAL component.
+
+    Notes
+    -----
+
+    """
+
+    def __init__(self, index, add_remotes: bool = True):
+        super().__init__(
+            index=index,
+            descr="Perform an ATCS offset",
+        )
+
+        atcs_usage = None if add_remotes else ATCSUsages.DryTest
+
+        self.atcs = ATCS(domain=self.domain, intended_usage=atcs_usage, log=self.log)
+
+    @classmethod
+    def get_schema(cls):
+        schema_yaml = """
+            $schema: http://json-schema.org/draft-07/schema#
+            $id: https://github.com/lsst-ts/ts_standardscripts/auxtel/offset_atcs.yaml
+            title: OffsetATCS v1
+            description: Configuration for OffsetATCS Script.
+            type: object
+            properties:
+              z:
+                type: number
+                description: Offset hexapod in z axis (mm).
+              x:
+                type: number
+                description: Offset hexapod in x axis (mm).
+              y:
+                type: number
+                description: Offset hexapod in y axis (mm).
+              u:
+                type: number
+                description: Rx offset (deg).
+              v:
+                type: number
+                description: Ry offset (deg).
+              m1:
+                type: number
+                description: M1 pressure offset (Pa).
+                maximum: 0
+              offset_telescope:
+                type: bool
+                description: When correcting coma, also offset the telescope?
+                type: boolean
+                default: true
+              reset_offsets:
+                type: array
+                description: >-
+                    List of axes to reset. Must be one of z, x, y, z, u, v,
+                    m1, all
+                items:
+                    type: string
+
+            additionalProperties: false
+            anyOf:
```

Comment:
Why only allow one offset at a time? I can see a reason to either allow offsetting or resetting, but one should be able to do more than one offset at the same time.

Reply:
I think the anyOf call here is saying we require at a minimum one of the following parameters be set. So you could still do an x,y,z offset in one. This case is actually covered in the unit test.

Reply:
ah! `anyOf`, for some reason I thought it was `oneOf`, my bad.